catalog/files/impl/src/test/java/org/projectnessie/catalog/files/s3/TestS3SessionsManager.java

```diff
@@ -16,283 +16,56 @@
-                    .policy(
-                        "{ \"Version\":\"2012-10-17\",\n"
-                            + "  \"Statement\": [\n"
-                            + "    {\"Effect\":\"Allow\", \"Action\": [ \"s3:GetObject\", \"s3:GetObjectVersion\", \"s3:s3:PutObject\", \"s3:DeleteObject\" ], \"Resource\": [ \"arn:aws:s3:::*},\n"
-                            + "   \n"
-                            + "}\n")
-                    .build())
-            .build();
-    String policy = S3SessionsManager.locationDependentPolicy(locations, options);
-    soft.assertThatThrownBy(() -> new ObjectMapper().readValue(policy, ObjectNode.class))
-        .isInstanceOf(IOException.class);
-  }
-
-  @ParameterizedTest
-  @MethodSource
-  void invalidSessionPolicyStatement(String invalid) {
-    StorageUri location = StorageUri.of("s3://foo/bar");
-    StorageLocations locations =
-        StorageLocations.storageLocations(StorageUri.of("s3://foo/"), List.of(location), List.of());
-    S3BucketOptions options =
-        ImmutableS3NamedBucketOptions.builder()
-            .clientIam(
-                ImmutableS3ClientIam.builder().enabled(true).statements(List.of(invalid)).build())
-            .build();
-    soft.assertThatThrownBy(() -> S3SessionsManager.locationDependentPolicy(locations, options))
-        .isInstanceOf(RuntimeException.class)
-        .cause()
-        .isInstanceOf(IOException.class);
-  }
-
-  static Stream<String> invalidSessionPolicyStatement() {
-    return Stream.of(
-        "\"Effect\":\"Deny\", \"Action\":\"s3:*\", \"Resource\":\"arn:aws:s3:::*/blockedNamespace/*\"}",
-        "\"Effect:\"Deny\", \"Action\":\"s3:*\", \"Resource\":\"arn:aws:s3:::*/blockedNamespace/*\"}",
-        "}\"Effect:\"Deny\", \"Action\":\"s3:*\", \"Resource\":\"arn:aws:s3:::*/blockedNamespace/*\"}");
-  }
-
   @Test
   void testExpiration() {
     AtomicLong time = new AtomicLong();
 
-    AtomicInteger clientCounter = new AtomicInteger();
-    Function<StsClientKey, StsClient> clientBuilder =
-        (parameters) -> {
-          clientCounter.incrementAndGet();
-          return Mockito.mock(StsClient.class);
-        };
-
     AtomicInteger counter = new AtomicInteger();
     AtomicReference<Credentials> credentials = new AtomicReference<>();
-    SessionCredentialsFetcher loader =
-        (client, key, duration, location) -> {
-          counter.incrementAndGet();
-          return credentials.get();
-        };
+    StsCredentialsFetcher loader = new MockStsCredentialsFetcher(credentials::get, counter);
 
     S3SessionsManager manager =
-        new S3SessionsManager(s3options, time::get, null, clientBuilder, Optional.empty(), loader);
+        new S3SessionsManager(10, Duration.ofMillis(10), loader, time::get, Optional.empty());
     S3BucketOptions options =
         ImmutableS3NamedBucketOptions.builder()
             .region("R1")
-            .clientIam(ImmutableS3ClientIam.builder().enabled(true).assumeRole("role").build())
```

Comment:
I think there was a bug here because we are building options with client IAM, but the tests below call `sessionCredentialsForServer`.